fontbe/src/glyphs.rs

```diff
@@ -278,3 +377,170 @@ fn path_el_type(el: &PathEl) -> &'static str {
+                        return None; // maybe next time?
+                    };
+
+                    // The transform we get here has changed because it got turned into F2Dot14 and i16 parts
+                    // We could go get the "real" transform from IR...?
+                    let affine = affine_for(c);
+                    let transformed_box = affine.transform_rect_bbox(bbox2rect(bbox.unwrap()));
+                    Some(rect2bbox(transformed_box))
+                })
+                .collect();
+            if missing_boxes {
+                trace!("bbox for {glyph_name} not yet resolveable");
+                continue;
+            }
+
+            let bbox = boxes
+                .into_iter()
+                .reduce(|acc, e| Bbox {
+                    x_min: cmp::min(acc.x_min, e.x_min),
+                    y_min: cmp::min(acc.y_min, e.y_min),
+                    x_max: cmp::max(acc.x_max, e.x_max),
+                    y_max: cmp::max(acc.y_max, e.y_max),
+                })
+                .unwrap();
+            trace!("bbox for {glyph_name} {bbox:?}");
+            bbox_acquired.insert(glyph_name.clone(), bbox);
+        }
+
+        // Kerplode if we didn't make any progress this spin
+        composites.retain(|(gn, _)| !bbox_acquired.contains_key(gn));
+        if pending == composites.len() {
+            panic!("Unable to make progress on composite bbox, stuck at\n{composites:?}");
+        }
+    }
+
+    // It'd be a shame to just throw away those nice boxes
+    for (glyph_name, bbox) in bbox_acquired.into_iter() {
+        let mut glyph = (*context.get_glyph(&glyph_name)).clone();
+        let Glyph::Composite(composite) = &mut glyph else {
+            panic!("{glyph_name} is not a composite; we shouldn't be trying to update it");
+        };
+        composite.bbox = bbox;
+        context.set_glyph(glyph_name, glyph);
+    }
+
+    Ok(())
+}
+
+impl Work<Context, Error> for GlyphMergeWork {
+    /// Generate [glyf](https://learn.microsoft.com/en-us/typography/opentype/spec/glyf)
+    /// and [loca](https://learn.microsoft.com/en-us/typography/opentype/spec/loca).
+    ///
+    /// We've already generated all the binary glyphs so all we have to do here is glue everything together.
+    fn exec(&self, context: &Context) -> Result<(), Error> {
+        compute_composite_bboxes(context)?;
+
+        let static_metadata = context.ir.get_static_metadata();
+        let glyph_order = &static_metadata.glyph_order;
+
+        // Glue together glyf and loca
+        // This isn't overly memory efficient but ... fonts aren't *that* big (yet?)
+        let mut loca = vec![0];
+        let mut glyf: Vec<u8> = Vec::new();
+        glyf.reserve(1024 * 1024);
+        glyph_order
+            .iter()
+            .map(|gn| context.get_glyph(gn))
+            .for_each(|g| {
+                let bytes = g.to_bytes();
+                loca.push(loca.last().unwrap() + bytes.len() as u32);
```

Comment:
as you know, `loca` has two version a "short" and a "long" one (with head.indexToLocFormat storing the loca format), you seem to unconditionally opt for the latter; at least add a TODO so we can later on implement the "short" variant

Reply:
I know :) I should have put a todo, adding now.

Reply:
Oh sorry, that's entirely intentional, the possible shrinking of loca is supposed to occur during the final merge. Clearly need a comment.